nix/modules/project/settings/default.nix

```diff
@@ -88,6 +96,6 @@ in
               fns
           );
       in
-      lib.mapAttrs applySettingsFor project.config.settings;
+      lib.mapAttrs applySettingsFor (lib.recursiveUpdate emptyPackageSettings project.config.settings);
```

Comment:
Why do we need `lib.recursiveUpdate emptyPackageSetting`?

Reply:
It considers the packages that are mentioned in `haskellProjects.<prj-name>.packages` but not in `haskellProjects.<prj-name>.settings`. i should include a test that would cover this case. Also, is the name misleading? Anything better you suggest?

Reply:
The `settings` option already does this:

https://github.com/srid/haskell-flake/blob/847292fc793a5c15c873e52e7751ee4267ef32a0/nix/modules/project/settings/default.nix#L21-L27

So, `project.config.settings` is already polyfilled with packages from `project.config.packages`.